Makefile

```diff
@@ -19,16 +19,38 @@ REPO_ROOT:=$(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
 
 all: test
 
+GIT_VERSION=$(shell git describe --tags --always --dirty)
+GIT_HASH ?= $(shell git rev-parse HEAD)
+DATE_FMT = +'%Y-%m-%dT%H:%M:%SZ'
+SOURCE_DATE_EPOCH ?= $(shell git log -1 --pretty=%ct)
+ifdef SOURCE_DATE_EPOCH
+    BUILD_DATE ?= $(shell date -u -d "@$(SOURCE_DATE_EPOCH)" "$(DATE_FMT)" 2>/dev/null || date -u -r "$(SOURCE_DATE_EPOCH)" "$(DATE_FMT)" 2>/dev/null || date -u "$(DATE_FMT)")
+else
+    BUILD_DATE ?= $(shell date "$(DATE_FMT)")
+endif
+GIT_TREESTATE = "clean"
+DIFF = $(shell git diff --quiet >/dev/null 2>&1; if [ $$? -eq 1 ]; then echo "1"; fi)
+ifeq ($(DIFF), 1)
+    GIT_TREESTATE = "dirty"
+endif
+
+PKG=sigs.k8s.io/promo-tools/v3/internal/version
+EXTRA_LDFLAGS='"-X $(PKG).gitVersion=$(GIT_VERSION) -X $(PKG).gitCommit=$(GIT_HASH) -X $(PKG).gitTreeState=$(GIT_TREESTATE) -X $(PKG).buildDate=$(BUILD_DATE)"'
+
+.PHONY: kpromo
+kpromo:
+	go build -trimpath -ldflags '-s -w -buildid= -extldflags "-static" $(EXTRA_LDFLAGS)' -o kpromo ./cmd/kpromo
```

Comment:
```suggestion
	go build -trimpath -ldflags '-s -w -buildid= -extldflags "-static" $(LDFLAGS)' -o kpromo ./cmd/kpromo
```

Reply:
Should we add separate targets for `gh2gcs` and `cip-mm` as well?

Reply:
will do

Reply:
thanks for the review and insights @justaugustus did all the changes PTAL